From: Olivier Fourdan <ofourdan@redhat.com>
Date: Mon, 7 Aug 2023 15:31:40 +0200
Subject: settings: Set xwayland_allow_byte_swapped_clients at startup

The value for the key "xwayland-allow-byte-swapped-clients" was being
updated on change, but never actually set at startup.

As a result, Xwayland was spawned with byteswap clients disabled, even
if the key was set to TRUE.

(cherry picked from commit d291d4a733793f9683199b22dfab1a255bfb8e8a)

Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/2938
Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3156
Applied-upstream: 44.4, commit:229d37f8613a9067131c765d3ff1a14339a8c215
---
 src/backends/meta-settings.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/backends/meta-settings.c b/src/backends/meta-settings.c
index 229b7f6..e3ebfe6 100644
--- a/src/backends/meta-settings.c
+++ b/src/backends/meta-settings.c
@@ -569,6 +569,7 @@ meta_settings_init (MetaSettings *settings)
   update_xwayland_allow_grabs (settings);
   update_xwayland_disable_extensions (settings);
   update_privacy_settings (settings);
+  update_xwayland_allow_byte_swapped_clients (settings);
 }
 
 static void
